test/AutoRest.TestServer.Tests/Mgmt/Unit/StringExtensionsTests.cs

```diff
@@ -17,5 +17,15 @@ public void ValidateResourceNameToPlural(string resourceName, string expected)
             var result = resourceName.ResourceNameToPlural();
             Assert.AreEqual(expected, result);
         }
+
+        [TestCase("MetadataRole", "MetadataRoleResource")]
+        [TestCase("PrivateResource", "PrivateResource")]
+        [TestCase("PrivateResource2", "PrivateResource2Resource")]
```

Comment:
Here is an open question: how do we deal with `XXXResourceYYY`? I think `PrivateResource2` is an extreme example which should not happen in reality. So maybe we just stick to simply appending a `Resource` suffix if not exists?

Reply:
This particular case is because we are making up names in our test project. Feel free to change the name of `PrivateResource2` if it bothers you

But it could happen in real life cases because we have the case of "extension resource" - multiple resources with same definition but different parents, and in this case, the parent's name will prepend to the name of this resource. We should make sure in this case, the `Resource` suffix on parent's resource name does not appear in the middle of child resource.

The RP `appservice` has this kind of cases a lot, please try to generate appservice to see what will happen